md/api.py

```diff
@@ -0,0 +1,138 @@
+from django.db import connections
+from django.db.models import Count, Q
+
+from rest_framework import viewsets
+from rest_framework.decorators import detail_route
+from rest_framework.response import Response
+from rest_framework_extensions.cache.decorators import cache_response
+from rest_framework_extensions.key_constructor import bits
+from rest_framework_extensions.key_constructor.constructors import DefaultObjectKeyConstructor
+
+from md.models import Agency, Stop
+from md import serializers
+from md.utils import GroupedData
+
+
+GROUPS = {'A': 'ASIAN',
+          'B': 'BLACK',
+          'I': 'NATIVE AMERIOCAN',
+          'U': 'OTHER',
+          'W': 'WHITE',
+          'H': 'HISPANIC'}
+
+# PURPOSE_CHOICES = {1: 'Speed Limit Violation',
+#                    2: 'Stop Light/Sign Violation',
+#                    3: 'Driving While Impaired',
+#                    4: 'Safe Movement Violation',
+#                    5: 'Vehicle Equipment Violation',
+#                    6: 'Vehicle Regulatory Violation',
+#                    7: 'Seat Belt Violation',
+#                    8: 'Investigation',
+#                    9: 'Other Motor Vehicle Violation',
+#                    10: 'Checkpoint'}
+
+GROUP_DEFAULTS = {'ASIAN': 0,
+                  'BLACK': 0,
+                  'NATIVE AMERICAN': 0,
+                  'OTHER': 0,
+                  'WHITE': 0,
+                  'HISPANIC': 0,
+                  'UNKNOWN': 0,
+                  }
```

Comment:
This has both 'UNKNOWN' and 'OTHER', which doesn't seem intentional????

If this should have a set of keys which matches the set of values of GROUPS, you can create this dictionary with ``GROUP_DEFAULTS = {k: 0 for k in GROUPS.values()}``.